core/src/main/java/io/grpc/internal/AbstractClientStream2.java

```diff
@@ -384,6 +384,7 @@ public void dispose() {
       headers = null;
     }
 
+    // This framer sends everything in the header. So the methods below do not apply. 
```

Comment:
What does writeHeaders have to do with whether this does compression?

For compression, you could say that the GET encoding we have chosen doesn't support compression.

For max outbound message size, it seems maybe add a TODO?